File: Frameworks/react-web/tests/unit/models/ods-menu-item.test.ts
import { describe, it, expect } from 'vitest'
import { parseMenuItem } from '../../../src/models/ods-menu-item.ts'

// ===========================================================================
// OdsMenuItem model tests
// ===========================================================================

describe('parseMenuItem', () => {
  // -------------------------------------------------------------------------
  // Basic parsing
  // -------------------------------------------------------------------------

  it('parses label and mapsTo', () => {
    const item = parseMenuItem({ label: 'Dashboard', mapsTo: 'dashboardPage' })
    expect(item.label).toBe('Dashboard')
    expect(item.mapsTo).toBe('dashboardPage')
  })

  it('parses roles array', () => {
    const item = parseMenuItem({
      label: 'Admin',
      mapsTo: 'adminPage',
      roles: ['admin', 'manager'],
    })
    expect(item.roles).toEqual(['admin', 'manager'])
  })

  // -------------------------------------------------------------------------
  // Missing / empty roles
  // -------------------------------------------------------------------------

  it('missing roles is undefined', () => {
    const item = parseMenuItem({ label: 'Home', mapsTo: 'homePage' })
    expect(item.roles).toBeUndefined()
  })

  it('handles empty roles array', () => {
    const item = parseMenuItem({ label: 'Home', mapsTo: 'homePage', roles: [] })
    expect(item.roles).toEqual([])
  })
})

File: Frameworks/flutter-local/test/models/ods_menu_item_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:ods_flutter_local/models/ods_menu_item.dart';

void main() {
  // =========================================================================
  // fromJson
  // =========================================================================

  group('OdsMenuItem.fromJson', () {
    test('parses label and mapsTo', () {
      final item = OdsMenuItem.fromJson({
        'label': 'Dashboard',
        'mapsTo': 'page_dashboard',
      });
      expect(item.label, 'Dashboard');
      expect(item.mapsTo, 'page_dashboard');
    });

    test('parses roles list', () {
      final item = OdsMenuItem.fromJson({
        'label': 'Admin Panel',
        'mapsTo': 'page_admin',
        'roles': ['admin', 'superuser'],
      });
      expect(item.roles, ['admin', 'superuser']);
    });

    test('missing roles is null', () {
      final item = OdsMenuItem.fromJson({
        'label': 'Home',
        'mapsTo': 'page_home',
      });
      expect(item.roles, isNull);
    });

    test('empty roles list', () {
      final item = OdsMenuItem.fromJson({
        'label': 'About',
        'mapsTo': 'page_about',
        'roles': <String>[],
      });
      expect(item.roles, isEmpty);
    });
  });
}
